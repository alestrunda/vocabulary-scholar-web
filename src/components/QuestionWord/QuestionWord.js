import React from 'react';
import PropTypes from 'prop-types';
import Hint from '../Hint';

class QuestionWord extends React.Component {
  componentDidMount() {
    this.handleQuestionReady();
  }

  componentDidUpdate(prevState) {
    if (this.props.word !== prevState.word) this.handleQuestionReady();
  }

  handleQuestionReady() {
    if (!this.props.word) return;
    this.props.onQuestionReady && this.props.onQuestionReady();
  }

  render() {
    return (
      <div className={this.props.className} style={this.props.style}>
        {this.props.word ? (
          <React.Fragment>
            <p>{this.props.word}</p>
            <Hint
              className="mt10"
              hint={this.props.hint}
              hintType={this.props.hintType}
              wordID={this.props.wordID}
            />
          </React.Fragment>
        ) : (
          <p className="text-no-data">Not available</p>
        )}
      </div>
    );
  }
}

QuestionWord.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  word: PropTypes.string,
  wordID: PropTypes.string,
  onQuestionReady: PropTypes.func,
  hint: PropTypes.string,
  hintType: PropTypes.string,
};

export default QuestionWord;
