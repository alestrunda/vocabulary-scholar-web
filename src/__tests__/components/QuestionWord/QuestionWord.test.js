import React from 'react';
import { shallow } from 'enzyme';

import QuestionWord from '../../../components/QuestionWord';

it('renders not available', () => {
  const root = shallow(<QuestionWord word="" wordID="school" />);
  expect(root).toMatchSnapshot();
});

it('renders correctly', () => {
  const root = shallow(<QuestionWord word="School" wordID="school" />);
  expect(root).toMatchSnapshot();
});
