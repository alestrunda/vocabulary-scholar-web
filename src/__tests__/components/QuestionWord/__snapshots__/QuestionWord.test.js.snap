// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionWord
    word="School"
    wordID="school"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <React.Fragment>
        <p>
          School
        </p>
        <Hint
          className="mt10"
          wordID="school"
        />
      </React.Fragment>,
      "className": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <p>
            School
          </p>,
          <Hint
            className="mt10"
            wordID="school"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "School",
          },
          "ref": null,
          "rendered": "School",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "mt10",
            "hint": undefined,
            "hintType": undefined,
            "wordID": "school",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <React.Fragment>
          <p>
            School
          </p>
          <Hint
            className="mt10"
            wordID="school"
          />
        </React.Fragment>,
        "className": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <p>
              School
            </p>,
            <Hint
              className="mt10"
              wordID="school"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "School",
            },
            "ref": null,
            "rendered": "School",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "mt10",
              "hint": undefined,
              "hintType": undefined,
              "wordID": "school",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`renders not available 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionWord
    word=""
    wordID="school"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="text-no-data"
      >
        Not available
      </p>,
      "className": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Not available",
        "className": "text-no-data",
      },
      "ref": null,
      "rendered": "Not available",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="text-no-data"
        >
          Not available
        </p>,
        "className": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Not available",
          "className": "text-no-data",
        },
        "ref": null,
        "rendered": "Not available",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
